### PR TITLE
[aws][feat] Add tag deletion/updating and resource deletion for backup service

### PR DESCRIPTION
**Pull request by 1101-1**
# Description

# To-Dos

- [x] I have created tests for any new or updated functionality.
- [x] I ran `tox` successfully.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://inventory.fix.security/code-of-conduct).

### CODE REVIEW

File: plugins/aws/fix_plugin_aws/resource/backup.py [hunk 1]
```diff
@@ -121,7 +123,7 @@ def add_tags(backup_job: AwsBackupJob) -> None:
                 "list-tags",
                 "Tags",
                 expected_errors=["ResourceNotFoundException"],
-                ResourceArn=backup_job.arn,
+                ResourceArn=backup_job.recovery_point_arn,
```
Comment (aquamatthias) [inline plugins/aws/fix_plugin_aws/resource/backup.py:126]: You would query for tags of the recovery point - that's wrong isn't it?
Does a job even have tags?

File: plugins/aws/fix_plugin_aws/resource/backup.py [hunk 2]
```diff
@@ -558,6 +656,43 @@ def called_collect_apis(cls) -> List[AwsApiSpec]:
...
+
+    def delete_resource_tag(self, client: AwsClient, key: str) -> bool:
+        client.call(
+            aws_service=self.api_spec.service,
+            action="untag-resource",
+            result_name=None,
+            ResourceArn=self.arn,
+            TagKeyList=[key],
+        )
+        return True
```
Comment (aquamatthias) [inline plugins/aws/fix_plugin_aws/resource/backup.py:685]: can we extract a separate base class? Seems to be repeated.
See CloudwatchTaggable as example

File: plugins/aws/fix_plugin_aws/resource/backup.py [hunk 3]
```diff
@@ -771,7 +972,7 @@ def add_tags(restore_job: AwsBackupRestoreJob) -> None:
                 "list-tags",
                 "Tags",
                 expected_errors=["ResourceNotFoundException"],
-                ResourceArn=restore_job.arn,
+                ResourceArn=restore_job.recovery_point_arn,
```
Comment (aquamatthias) [inline plugins/aws/fix_plugin_aws/resource/backup.py:975]: Same here.
You would query for tags of the recovery point - that's wrong isn't it?
Does a job even have tags?

File: plugins/aws/fix_plugin_aws/resource/backup.py [hunk 4]
```diff
@@ -868,7 +1069,7 @@ def add_tags(copy_job: AwsBackupCopyJob) -> None:
                 "list-tags",
                 "Tags",
                 expected_errors=["ResourceNotFoundException"],
-                ResourceArn=copy_job.arn,
+                ResourceArn=copy_job.source_recovery_point_arn,
```
Comment (aquamatthias) [inline plugins/aws/fix_plugin_aws/resource/backup.py:1072]: same here.

File: plugins/aws/fix_plugin_aws/resource/backup.py (hunk 1 above)
Comment (1101-1) [reply plugins/aws/fix_plugin_aws/resource/backup.py:126]: Tags from the original resource are automatically copied to the recovery points created during backup jobs. This ensures that the backup data retains the tagging information of the original resources.

https://docs.aws.amazon.com/aws-backup/latest/devguide/tags-on-backups.html

But yes, it is not a backup job/copy job/restore job tags

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**